File: backend/src/main/java/com/edushare/backend/service/CommentService.java
package com.edushare.backend.service;

import com.edushare.backend.model.Comment;
import com.edushare.backend.model.Post;
import com.edushare.backend.repository.CommentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class CommentService {

    private final CommentRepository commentRepository;

    @Autowired
    public CommentService(CommentRepository commentRepository) {
        this.commentRepository = commentRepository;
    }
    @Autowired
    private PostService postService;
    
    // Create comment
    public Comment createComment(Comment comment) {
        comment.setCreatedAt(LocalDateTime.now());
        comment.setUpdatedAt(LocalDateTime.now());
        return commentRepository.save(comment);
    }

    // Get all comments
    public List<Comment> getAllComments() {
        return commentRepository.findAll();
    }

    // Get comment by ID
    public Optional<Comment> getCommentById(String id) {
        return commentRepository.findById(id);
    }

    // Get comments by post
    public List<Comment> getCommentsByPostId(String postId) {
        return commentRepository.findByPostId(postId);
    }

    // Get comments by user
    public List<Comment> getCommentsByUserId(String userId) {
        return commentRepository.findByUserId(userId);
    }

    // Update comment
    public Comment updateComment(Comment comment) {
        comment.setUpdatedAt(LocalDateTime.now());
        return commentRepository.save(comment);
    }

    // Delete comment
    public void deleteComment(String id) {
        commentRepository.deleteById(id);
    }

    public String findPostOwnerId(String postId) {
    return postService.getPostById(postId)
            .map(Post::getUserId)
            .orElseThrow(() -> new RuntimeException("Post not found for ID: " + postId));
}

}

File: frontend/src/Service/CommentService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/comments';

export const createComment = async (commentData) => {
  try {
    const response = await axios.post(API_BASE_URL, commentData);
    return response.data;
  } catch (error) {
    console.error('Error creating comment:', error);
    throw error;
  }
};

export const getCommentsByPostId = async (postId) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/post/${postId}`);
    return response.data._embedded?.commentList || [];
  } catch (error) {
    console.error('Error fetching comments:', error);
    throw error;
  }
};

export const updateComment = async (id, content) => {
  try {
    const response = await axios.put(`${API_BASE_URL}/${id}`, { content });
    return response.data;
  } catch (error) {
    console.error('Error updating comment:', error);
    throw error;
  }
};

export const deleteComment = async (id) => {
  try {
    const response = await axios.delete(`${API_BASE_URL}/${id}`);
    return response.data;
  } catch (error) {
    console.error('Error deleting comment:', error);
    throw error;
  }
};

export const getCommentsByUserId = async (userId) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/user/${userId}`);
    return response.data._embedded?.commentList || [];
  } catch (error) {
    console.error('Error fetching user comments:', error);
    throw error;
  }
};